src/java.base/share/classes/sun/net/www/protocol/http/DigestAuthentication.java

```diff
@@ -64,41 +71,35 @@
 
     private String authMethod;
 
-    private static final String secPropName =
-        "jdk.httpdigest.defaultDisabledAlgorithms";
+    private static final String propPrefix = "http.auth.digest.";
 
-    private static final String compatPropName = "http.auth.digest." +
+    private static final String compatPropName = propPrefix +
         "quoteParameters";
 
-    // default set of disabled message digest algorithms that may not be
-    // used for proxy connections, or plain text http server connections
-
-    private static final Set<String> defDisabledAlgs = getDefaultAlgs();
-
-    private static Set<String> getDefaultAlgs() {
-        Set<String> disabledAlgs = new HashSet<>();
-        @SuppressWarnings("removal")
-        String secprops = AccessController.doPrivileged(
-            new PrivilegedAction<>() {
-                public String run() {
-                    return Security.getProperty(secPropName)
-                                   .replaceAll("\\s", "")
-                                   .toUpperCase();
-                }
-            }
-        );
-        String[] algorithms = secprops.split(",");
-        for (String algorithm : algorithms) {
-            disabledAlgs.add(algorithm);
+    // Takes a set and input string containing comma separated values. converts to upper
+    // case, and trims each value, then applies given function to set and value
+    // (either add or delete element from set)
+    private static void processPropValue(String input,
+        Set<String> theSet,
+        BiConsumer<Set<String>,String> consumer)
+    {
+        if (input == null) {
+            return;
+        }
+        String[] values = input.toUpperCase(Locale.ROOT).split(",");
+        for (String v : values) {
+            consumer.accept(theSet, v.trim());
         }
-        return Collections.unmodifiableSet(disabledAlgs);
     }
 
+    private static final String secPropName =
+        propPrefix + "disabledAlgorithms";
+
     // A net property which overrides the disabled set above.
     private static final String enabledAlgPropName =
-        "http.auth.digest.enabledAlgorithms";
+        propPrefix + "reEnabledAlgorithms";
 
-    private static final Set<String> disabledAlgs = new HashSet<>();
+    private static final Set<String> disabledAlgorithms = new HashSet<>();
```

Comment:
The Set is private to the class and is not modified after the static initializer completes.  It's not clear to me how using Set.copyOf provides strong MT-safe guarantees than this.

Reply:
Better safe than sorry. An alternative could be to use ConcurrentHashMap.newKeySet(); But since it's supposed to be both immutable and MT-safe then Set.copyOf() would probably be a lighter and better choice.